src/test/java/org/apache/commons/net/SubnetUtilsTest.java

```diff
@@ -375,4 +375,52 @@ public void testNET641() {
         assertFalse(new SubnetUtils("192.168.1.0/31").getInfo().isInRange("0.0.0.0"));
         assertFalse(new SubnetUtils("192.168.1.0/32").getInfo().isInRange("0.0.0.0"));
     }
+
+    /**
+     * Test case for IPv6 addresses
+     *
+     * Relate to NET-405
+     */
+    public void testIP6Address() {
+        //Valid address test
+        SubnetUtils subnetUtils = new SubnetUtils("2001:db8:3c0d:5b6d:0:0:42:8329/58");
+        SubnetInfo subnetInfo = subnetUtils.getInfo();
+        assertEquals("CIDR-Notation", "2001:db8:3c0d:5b6d::42:8329/58", subnetInfo.getCIDRNotation());
+        assertEquals("Lowest Address", "2001:db8:3c0d:5b40::", subnetInfo.getLowAddress());
+        assertEquals("Highest Address", "2001:db8:3c0d:5b7f:ffff:ffff:ffff:ffff", subnetInfo.getHighAddress());
+        assertEquals("Address counts", "1180591620717411303424", subnetInfo.getAddressCountString());
+
+        //Address range test
+        assertTrue(subnetInfo.isInRange("2001:db8:3c0d:5b6d:0:0:42:8329"));
+        assertTrue(subnetInfo.isInRange("2001:db8:3c0d:5b40::"));
+        assertTrue(subnetInfo.isInRange("2001:db8:3c0d:5b7f:ffff:ffff:ffff:ffff"));
+        assertTrue(subnetInfo.isInRange("2001:db8:3c0d:5b53:0:0:0:1"));
+        assertFalse(subnetInfo.isInRange("2001:db8:3c0d:5b3f:ffff:ffff:ffff:ffff"));
+        assertFalse(subnetInfo.isInRange("2001:db8:3c0d:5b80::"));
+
+        //Invalid address test
+        try {
+            //No colons
+            new SubnetUtils("2001db83c0d5b6d004283291/58");
+
+            //Out of Range
+            new SubnetUtils("2001d:b83c0:d5b6d:2428:3291f:b8:b75fe:abef5/58");
+
+            //Beginning a Colon
+            new SubnetUtils(":2001:db8:0:2:0:9abc/58");
+
+            //Zero Compression
+//            subnetUtils = new SubnetUtils("2001:db8::2::9abc/58");
+
+            //Length
+            new SubnetUtils("2001:db8:3c0d:5b6d:0:0:42:8329:1/58");
+
+            //Out of Subnet
+            new SubnetUtils("2001:db8:3c0d:5b6d:0:0:42:8329/129");
+
+            fail("Should have thrown IllegalArgumentException");
+        } catch (IllegalArgumentException expected) {
+            // Ignored
```

Comment:
Only one test please per try/catch clause.
I assume the intent is to check that they must all throw IAE
As it stands, only one statement has to throw IAE for the entire block to succeed.

Better yet, use separate test cases for each.
Otherwise only the first failure will be detected.

Reply:
I divided test cases for each test category.

Reply:
I will modify it but take time.